### PR TITLE
upgrade azure sdk and get linuxFxVersion using sdk

### PR DESCRIPTION
**Pull request by juniwang**
The newer version(1.3.0) of azure sdk fixed the `linuxFxVersion` issue. The PR upgraded the azure sdk and made changes correspondingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -97,6 +97,12 @@
 
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
+            <artifactId>credentials</artifactId>
```
Comment (ArchangelSDY) [inline pom.xml]: Why adding this dependency?

Comment (juniwang) [reply pom.xml]: Error class `SecretBytes` NotFound thrown on my local env. I fixed by adding the dependency. Any other way to fix it?

Comment (ArchangelSDY) [reply pom.xml]: Not sure if `SecretBytes` is required by this plugin or azure-credentials plugin. If the latter case, maybe we should add it to the azure-credentials plugin?

Comment (juniwang) [reply pom.xml]: Your are right, we can add it to azure-credential later. I revoked the change. The exception was warning only. I can ignore it..

Comment (juniwang) [reply pom.xml]: Have to add it back. Otherwise it breaks CI.

**Pull request merged**